x/incentive/types/rewards.go

```diff
@@ -0,0 +1,87 @@
+package types
+
+import (
+	"fmt"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// RewardPeriod stores the state of an ongoing reward
+type RewardPeriod struct {
+	Denom         string        `json:"denom" yaml:"denom"`
+	Start         time.Time     `json:"start" yaml:"start"`
+	End           time.Time     `json:"end" yaml:"end"`
+	Reward        sdk.Coin      `json:"reward" yaml:"reward"` // per second reward payouts. For example, if we know from params that 10000KAVA is being paid out over 1 week (604800 rewards periods), then the value of reward would be (10000 * 1000000)/604800 = 16534ukava per second
+	ClaimEnd      time.Time     `json:"claim_end" yaml:"claim_end"`
+	ClaimTimeLock time.Duration `json:"claim_time_lock" yaml:"claim_time_lock"` // the amount of time rewards are timelocked once they are sent to users
+}
+
+// String implements fmt.Stringer
+func (rp RewardPeriod) String() string {
+	return fmt.Sprintf(`Reward Period:
+	Denom: %s,
+	Start: %s,
+	End: %s,
+	Reward: %s,
+	Claim End: %s,
+	Time Lock: %s`,
+		rp.Denom, rp.Start, rp.End, rp.Reward, rp.ClaimEnd, rp.ClaimTimeLock)
+}
+
+// NewRewardPeriod returns a new RewardPeriod
+func NewRewardPeriod(denom string, start time.Time, end time.Time, reward sdk.Coin, claimEnd time.Time, claimTimeLock time.Duration) RewardPeriod {
```

Comment:
Opinion: could condense this as
```suggestion
func NewRewardPeriod(denom string, reward sdk.Coin, start, end, claimEnd time.Time, claimTimeLock time.Duration) RewardPeriod {
```

Reply:
^^